inc/classes/optimization/CSS/class-combine-google-fonts.php

```diff
@@ -0,0 +1,171 @@
+<?php
+namespace WP_Rocket\Optimization\CSS;
+
+/**
+ * Combine Google Fonts
+ *
+ * @since 3.1
+ * @author Remy Perona
+ */
+class Combine_Google_Fonts {
+	/**
+	 * Crawler Instance
+	 *
+	 * @since 3.1
+	 * @author Remy Perona
+	 *
+	 * @var HtmlPageCrawler
+	 */
+	protected $crawler;
+
+	/**
+	 * Found fonts
+	 *
+	 * @since 3.1
+	 * @author Remy Perona
+	 *
+	 * @var array
+	 */
+	protected $fonts;
+
+	/**
+	 * Found subsets
+	 *
+	 * @since 3.1
+	 * @author Remy Perona
+	 *
+	 * @var array
+	 */
+	protected $subsets;
+
+	/**
+	 * Constructor
+	 *
+	 * @since 3.1
+	 * @author Remy Perona
+	 *
+	 * @param HtmlPageCrawler $crawler Crawler instance.
+	 */
+	public function __construct( $crawler ) {
+		$this->crawler = $crawler;
+		$this->fonts   = [];
+		$this->subsets = [];
+	}
+
+	/**
+	 * Combines multiple Google Fonts links into one
+	 *
+	 * @since 3.1
+	 * @author Remy Perona
+	 *
+	 * @return string
+	 */
+	public function optimize() {
+		$nodes = $this->find( 'link[href*="fonts.googleapis.com/css"]' );
+
+		if ( ! $nodes ) {
+			return $this->crawler->saveHTML();
+		}
+
+		$this->parse( $nodes );
+
+		if ( empty( $this->fonts ) ) {
+			return $this->crawler->saveHTML();
+		}
+
+		if ( ! $this->combine() ) {
+			return $this->crawler->saveHTML();
+		}
+
+		$nodes->remove();
+
+		return $this->crawler->saveHTML();
+	}
+
+	/**
+	 * Finds links to Google fonts
+	 *
+	 * @since 3.1
+	 * @author Remy Perona
+	 *
+	 * @param string $pattern Pattern to search for.
+	 * @return bool\HtmlPageCrawler
+	 */
+	protected function find( $pattern ) {
+		try {
+			$nodes = $this->crawler->filter( $pattern );
+		} catch ( Exception $e ) {
+			return false;
+		}
+
+		if ( $nodes->count() <= 1 ) {
+			return false;
+		}
+
+		return $nodes;
+	}
+
+	/**
+	 * Parses found nodes to extract fonts and subsets
+	 *
+	 * @since 3.1
+	 * @author Remy Perona
+	 *
+	 * @param HtmlageCrawler $nodes Found nodes for the pattern.
+	 * @return void
+	 */
+	protected function parse( $nodes ) {
+		$nodes->each( function( \Wa72\HtmlPageDom\HtmlPageCrawler $node, $i ) {
+			// Get fonts name.
+			$font = str_replace( array( '%7C', '%7c' ), '|', $node->attr( 'href' ) );
+			$font = explode( 'family=', $font );
+			$font = isset( $font[1] ) ? explode( '&', $font[1] ) : array();
```

Comment:
I think it should be `'&amp;'` instead of `'&'` here.
But anyway, I would do that instead:
```php
$font = wp_parse_url( $node->attr( 'href' ) );

if ( ! isset( $font['query'] ) ) {
	continue;
}

$font = html_entity_decode( $font );
$font = wp_parse_args( $font );
```

Now you have an array like this:
```
Array
(
    [family] => Open Sans:400,700|Roboto:400,700|Song Myung|Sunflower:300,500,700
    [subset] => latin-ext
)
```
Be careful though, now you have `Open Sans` instead of `Open+Sans`, so you'll probably need `htmlentities()` + `rawurlencode()` after that.